### PR TITLE
Add Support for disabled state tabs

### LINKED ISSUE #6162: Add disabled state for Tabs
Want to support a disabled state for tabs with the ability to attach a tooltip.

Core issue and designs: https://github.com/patternfly/patternfly/issues/4257

### PR DESCRIPTION
**Pull request by CooperRedhat**
**What**: Closes #6162 

**Additional issues**:
- Refactored Tab into it's own component in order to forwardRef for `Tooltip` in `aria-disabled` case re: [Original Request](https://github.com/patternfly/patternfly-design/issues/1066)
- Added disabled state examples to match core PR for same issue

### CODE REVIEW

**Review by mcarrano — DISMISSED** [review]
Looks good to me. Thanks @CooperRedhat !

**Review by nicolethoen — COMMENTED** [review]
LGTM After a11y tests pass

File: packages/react-core/src/components/Tabs/Tab.tsx [hunk 1]
```diff
@@ -18,8 +22,72 @@ export interface TabProps extends Omit<React.HTMLProps<HTMLAnchorElement | HTMLB
...
+      ref={innerRef}
+    >
+      <TabButton
+        className={css(
+          styles.tabsLink,
+          isDisabled && href && styles.modifiers.disabled,
+          isAriaDisabled && styles.modifiers.ariaDisabled
+        )}
+        disabled={isDisabled}
+        aria-disabled={isDisabled || isAriaDisabled}
```
Comment (mcoker) [inline packages/react-core/src/components/Tabs/Tab.tsx:76]: It doesn't hurt anything, but you don't need `aria-disabled="true"` when it's a `button` with the `disabled` attribute.

**Review by mcoker — CHANGES_REQUESTED** [review]
Looks good! Can you also add an example of when a tab button is an `<a>` (in the nav element examples - https://patternfly-react-pr-6261.surge.sh/components/tabs#using-the-nav-element)? Here are the core examples - https://pf4.patternfly.org/components/tabs#using-the-nav-element

File: packages/react-core/src/components/Tabs/Tab.tsx (hunk 1 above)
Comment (CooperRedhat) [reply packages/react-core/src/components/Tabs/Tab.tsx]: Thanks for pointing that out, fixed it and added disabled state to nav examples

**Comment by CooperRedhat** [comment]
Note: Nav examples with anchor tags/hrefs were not updating active state, added preventDefault to click handler (which is what jumplinks do), should be working now.

**Review by mcoker — CHANGES_REQUESTED** [review]
Just one change- with a disabled nav tab/link:
* Replace the `disabled` attribute with `aria-disabled="true"` (`disabled` isn't valid on an `<a>`)
* Add `tabindex="-1"` to remove it from the tab order

File: packages/react-core/src/components/Tabs/Tabs.tsx [hunk 2]
```diff
@@ -303,46 +312,7 @@ export class Tabs extends React.Component<TabsProps, TabsState> {
...
-                    tabContentRef={tabContentRef}
-                    ouiaId={childOuiaId}
-                    {...rest}
-                  >
-                    {title}
-                  </TabButton>
-                </li>
-              );
-            })}
+            {filteredChildren}
```
Comment (kmcfaul) [inline packages/react-core/src/components/Tabs/Tabs.tsx:315]: I'm a little confused why this mapping is no longer needed, did the structure of `filteredChildren` change?

**Review by kmcfaul — COMMENTED** [review]
Looks like the examples have some duplicate ids that are causing a11y tests to fail.

File: packages/react-core/src/components/Tabs/Tabs.tsx (hunk 2 above)
Comment (CooperRedhat) [reply packages/react-core/src/components/Tabs/Tabs.tsx:315]: The tab component used to return null, i.e. render nothing. So filtered children was just an array of objects. I needed a ref on the tab component for tooltips, so I had to forward ref and render an instance to attach a ref to.

**Comment by CooperRedhat** [comment]
> Looks like the examples have some duplicate ids that are causing a11y tests to fail.

Looks like the duplicate IDs came from wrapping a tab in a tooltip, we'll see if it passes.

**Review by nicolethoen — DISMISSED** [review]

**Review by mcoker — CHANGES_REQUESTED** [review]
Screen Shot 2021-09-10 at 1 52 42 PM

For regular disabled nav links:
* Remove `disabled`
  * An `<a>` doesn't support the `disabled` attr.
* Add `aria-disabled="true"`
  * Since the browser doesn't disable a link for you, you have to tell screen readers its disabled
* Add `tabindex="-1"`
  * Since its disabled, you don't want users to be able to tab to it (just like tab buttons that are `disabled`)

Screen Shot 2021-09-10 at 1 53 12 PM

For aria-disabled tab links:
* Add `aria-disabled="true"`
  * Same as above
* Remove the `tabindex` attr
  * You want users to be able to tab to it to trigger the popover/tooltip

**Review by tlabaj — CHANGES_REQUESTED** [review]
Can you had a demo and some integration test to verify the disables tabs?

**Review by jessiehuff — COMMENTED** [review]
So in this scenario, I don't think we actually want to add a `tabIndex="-1"` to the tab because it will make the tooltip completely inaccessible by keyboard. The weird thing about `aria-disabled` elements is they're kind of like a combination of disabled and interactive elements. Adding a tooltip makes it kind of interactive, but the main action of the tab is disabled. However, by playing with the preview, even when I take the `tabIndex` off the tooltip isn't displaying on focus so I'm not positive what's going on there. I also noticed that the tooltip isn't getting announced in VO. I think we need an `aria-describedby="whatever the tooltip's id is"`. A good example to follow is our [aria-disabled button with tooltip examples](https://www.patternfly.org/v4/components/button#aria-disabled-button-with-tooltip).

**Review by mcoker — DISMISSED** [review]
LGTM!

**Review by tlabaj — APPROVED** [review]

**Review by mcoker — APPROVED** [review]

**Review by kmcfaul — APPROVED** [review]

**Review by jessiehuff — COMMENTED** [review]
So I noticed in reviewing that the tooltip still isn't showing up on keyboard focus. I asked Nicole if she could test that aspect too, and she noticed that if you pressed enter the tooltip will appear. That's pretty unexpected that it would appear on hover for mouse users but require interaction to appear for keyboard users. However, we also noticed that this happened with the tooltip on enabled tabs as well. Nicole came up with a good theory that it may have to do with a discrepancy between where the tooltip is put (on the li) and where the focus is (on a button). We could probably create a follow up issue to investigate.

Also, I think we only need the `tabIndex="0"` if there's something interactive on the disabled tab. We should probably create an issue to add this to our documentation as well.

**Pull request merged**